volatility3/framework/plugins/linux/elfs.py

```diff
@@ -28,10 +36,94 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             requirements.ListRequirement(name = 'pid',
                                          description = 'Filter on specific process IDs',
                                          element_type = int,
-                                         optional = True)
+                                         optional = True),
+            requirements.BooleanRequirement(name = 'dump',
+                                            description = "Extract listed processes",
+                                            default = False,
+                                            optional = True)
         ]
 
+    @classmethod
+    def elf_dump(
+            cls, context: interfaces.context.ContextInterface,
+            layer_name: str,
+            elf_table_name: str,
+            vma: interfaces.objects.ObjectInterface,
```

Comment:
I generally prefer sending simple types in API function signatures, since volatility objects come with a lot of baggage so if we ever end up marshalling/unmarshalling calls to this function it might get sticky, but since we already pass `task` this way here (and in several places), it seems ok to allow it.  Just wanted to make sure it's more an exception than a rule...  5:)

Reply:
Mmmm, you're right. I could replace the `vma` and `task`-variables with `vma.vm_start`, `path`, `task.comm` and `task.pid` if you think that's better :) The only (minor) disadvantage is that the logic for retrieving/calculating these variables has to be placed before every call to this function (as far as these variables aren't already present at those places), causing (very) little code duplication.

Reply:
No, as I say, @atcuno convinced me that it makes sense for certain methods to take a `task` object rather than listing each and every individual value we wanted to use from them.  I think best to leave it as is, I just wanted to make sure that my reasoning was documented somewhere so I (or someone else) could go "ah ha! but it says here..." at some point in the future...  5;P